src/Gybs.Logic.Validation/Internal/ConfiguredValidationRule.cs

```diff
@@ -47,6 +47,13 @@ internal Task<IResult> ValidateAsync(IServiceProvider serviceProvider)
             t => t.rule.GetMethod(nameof(IValidationRule<object>.ValidateAsync), new[] { DataType! })!
```

Comment:
It should probably be fine as `GetMethod` should return public implementation only, but I would add a test scenario with class implementing both Task-based and ValueTask-based intefaces for the same data type.